ui/apps/dashboard/src/components/TimeRangeInput/TimeInput.tsx

```diff
@@ -148,11 +148,18 @@ export function TimeInput({ onChange, placeholder, required }: Props) {
   }
 
   function handleInputKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
-    if (event.code === 'Enter' && state.status === 'suggestion_available') {
-      event.preventDefault();
-      dispatch({ type: 'applied_suggestion' });
-      onChange(state.suggestedDateTime);
+    if (!(event.code === 'Enter' && state.status === 'suggestion_available')) {
+      return;
     }
+    event.preventDefault();
+    dispatch({ type: 'applied_suggestion' });
+    onChange(state.suggestedDateTime);
+  }
+
+  function handleSuggestionClick() {
+    if (state.status !== 'suggestion_available') return;
+    dispatch({ type: 'applied_suggestion' });
+    onChange(state.suggestedDateTime);
   }
```

Comment:
suggestion: re-use this code in both places as to avoid future drift or bugs where one is changesd, but not the other. For example: 
```ts
  function handleInputKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
    if (event.code === 'Enter') {
      event.preventDefault();
      applySuggestion();
    }
  }

  function applySuggestion() { // use as click handler as well
    if (state.status !== 'suggestion_available') return;
    dispatch({ type: 'applied_suggestion' });
    onChange(state.suggestedDateTime);
  }
```